src/Merk.php

```diff
@@ -15,9 +15,12 @@ public function __construct(Client $client)
         $this->client = $client;
     }
 
-    public function getCompanyByRegNo(string $regno): Response
+    public function getCompanyByRegNo(string $regno, string $countryCode = 'cz'): Response
```

Comment:
Není zde důvod nějakou zemi preferovat na úkor jiné. Knihovna je univerzální.

API knihovny by mělo opisovat API Merku – tedy umožnit nedefinovat zemi. Tato úprava by znemožnila poslání dotazu na Merk bez ohledu na zemi.

Dále vidím problém:
[Merk API](https://api.merk.cz/docs/#!/company/Companies_list) poskytuje v tomto endpointu více parametrů ([dokumentace](https://api.merk.cz/docs/#!/company/Companies_list)). Je skutečně navržená úprava rozumná z dlouhodobého hlediska? Parametr jsi přidal jen s vědomím, že jej zrovna potřebujeme. Ale knihovna je veřejná a její API by mělo být univerzální a nadčasové, aby odráželo možnosti Merk API jako takového – nikoliv naše aktuální potřeby. Pokud bychom to vedli takto, dostaneme se brzy do stavu, že budeme mít metody s hromadou argumentů, které jsou v pořadí, v jakém jsme parametry přidávali.

Reply:
Rozumím. Mám k tomu ale jednu výhradu. CountryCode je zásadní, pokud člověk chce číst z jejich SK databáze. Bez toho se knihovna neobejde. Ostatní parametry jsou opravdu volitelné. Hodnota CZ je navíc opravdu výchozí hodnotou a tak jsem to zde v kódu chtěl jasně definovat. Pokud hodnota nebude zadaná, tak se použije CZ. Mohlo by dojít k mýlce, že pokud hodnotu nepoužiji, tak budu číst z obou databází. Jak z CZ, tak z SK.

Reply:
Podle čeho víme, že je `cz` hodnota výchozí? Nevyplývá to ani z dokumentace, ani z API. Pak to vypadál že naše API mění povahu rozhraní. Chtělo by to dokomentovat.

Reply:
https://api.merk.cz/docs/#!/suggest/Suggest_list
![image](https://user-images.githubusercontent.com/4076655/140041200-4bdbd77f-66b6-4d42-b7dc-f6119a14640d.png)

Reply:
Ano, to je `/suggest/`. Ale tato diskuze probíhá u `/company/`, kde default není stanoven.
![image](https://user-images.githubusercontent.com/1657322/140041912-77c6dbb0-f0c4-4d81-8374-99539752e0b9.png)

Reply:
OK, po otestování tedy ok, CZ je fakt výchozí hodnota. 

Prosím o úpravy: 

- [ ] Odstranit `strtolower`, jedná se o `enum` hodnotu u které by SDK nemělo zasahovat do její podoby.
- [ ] Doplnit do kódu poznámku `'cz' value is default by Merk API`

Reply:
Upraveno.